### DISCUSSION TITLE
Clarification about `$schema` in embedded schemas

### DISCUSSION THREAD

**Discussion by santhosh-tekuri** [json-schema-org/community#707] [Q&A] [answered]
I am trying to implement mixed dialects support in my jsonschema library. and I want to clarify my understanding;

the 2012 specification says:
```
The "$schema" keyword SHOULD be used in the document root schema object, 
and MAY be used in the root schema objects of embedded schema resources. It 
MUST NOT appear in non-resource root schema objects. If absent from the 
document root schema, the resulting behavior is implementation-defined
```

my understanding is that: if an embedded schema has `$schema` then it MUST have `$id` attribute also. 

please correct me if my understanding is correct.

**Comment by notEthan** [comment, marked as answer]
That is correct (as I understand it).
